### PR TITLE
test: add proxy tests for clientv3 integration tests

### PR DESCRIPTION
**Pull request by heyitsanthony**

### CODE REVIEW

File: proxy/grpcproxy/watch_broadcast.go [hunk 1]
```diff
@@ -50,27 +50,24 @@ func newWatchBroadcast(wp *watchProxy, w *watcher, update func(*watchBroadcast))
 	wb.add(w)
 	go func() {
 		defer close(wb.donec)
-		// loop because leader loss will close channel
```
Comment (fanminshi) [inline proxy/grpcproxy/watch_broadcast.go]: i wonder what's the problem with the original loop implementation?

File: proxy/grpcproxy/watch_broadcast.go [hunk 2]
```diff
@@ -50,27 +50,24 @@ func newWatchBroadcast(wp *watchProxy, w *watcher, update func(*watchBroadcast))
...
+		opts := []clientv3.OpOption{
+			clientv3.WithRange(w.wr.end),
+			clientv3.WithProgressNotify(),
+			clientv3.WithRev(wb.nextrev),
+			clientv3.WithPrevKV(),
+		}
+		// The create notification should be the first response;
+		// if the watch is recreated following leader loss, it
+		// shouldn't post a second create response to the client.
+		if wb.responses == 0 {
```
Comment (fanminshi) [inline proxy/grpcproxy/watch_broadcast.go]: if the retry loop is gone, should we worry about the above situation?

File: proxy/grpcproxy/watch_broadcast.go (hunk 1 above)
Comment (heyitsanthony) [inline proxy/grpcproxy/watch_broadcast.go]: watches that don't need requireleader will always fail even if they're opened at historical revisions

File: proxy/grpcproxy/watch_broadcast.go (hunk 2 above)
Comment (heyitsanthony) [inline proxy/grpcproxy/watch_broadcast.go]: yeah this can be removed

**Comment by fanminshi** [comment]
lgtm

**Comment by xiang90** [comment]
lgtm.

**Pull request merged**